TSPL.docc/LanguageGuide/Concurrency.md

```diff
@@ -1591,6 +1591,370 @@ You can also use an unavailable conformance
+for global actor isolated types.
+The following code example declares a conformance to `Equatable`
+for a main-actor isolated class,
+and Swift infers main-actor isolation for the conformance:
+
+```swift
+@MainActor
+class Person {
+    var id: Int
+}
+
+// Inferred to be a @MainActor conformance to Equatable
+extension Person: Equatable {
+    static func ==(lhs: Person, rhs: Person) -> Bool {
+        return lhs.id == rhs.id
+    }
+}
+```
+
+You can opt out of this inference for a global-actor-isolated type
+by explicitly declaring that a protocol conformance is nonisolated.
+The following code example declares
+a nonisolated conformance to `Equatable` in an extension:
+
+```swift
+@MainActor
+class Person {
+    let id: Int
+}
+
+extension Person: nonisolated Equatable {
+    nonisolated static func ==(lhs: Person, rhs: Person) -> Bool {
+        return lhs.id == rhs.id
+    }
+}
+```
+
+### Data-Race Safety for Isolated Conformances
+
+Swift prevents data races for isolated conformances
+by ensuring that protocol requirements are only used
+on the global actor that the conformance is isolated to.
+In generic code,
+where the concrete conforming type is abstracted away,
+protocol requirements can be used through type parameters or `any` types.
+
+#### Using Isolated Conformances
+
+A conformance requirement to `Sendable`
+allows generic code to send parameter values to concurrently-executing code.
+If generic code accepts non-`Sendable` types,
+then the generic code can only use the input values
+from the current isolation domain.
+These generic APIs can safely accept isolated conformances
+and call protocol requirements
+as long as the caller is on the same global actor
+that the conformance is isolated to.
+The following code has a protocol `Dancer`,
+a class `Ballerina` with a main-actor isolated conformance to `Dancer`,
+and calls to the `Dancer.perform` requirement
+from a main-actor task and a concurrent task:
+
+```swift
+protocol Dancer {
+    func perform()
+}
+
+func startRecital(_ dancer: some Dancer) {
+    dancer.perform()
+}
+
+@MainActor class Ballerina: @MainActor Dancer { ... }
+
+Task { @MainActor in
+    let ballerina = Ballerina()
+    startRecital(ballerina)
+
+    let dancer: any Dancer = ballerina
+    dancer.perform()
+}
+
+Task { @concurrent in
+    let ballerina = Ballerina()
+    startRecital(c)  // Error
+
+    let dancer: any Dancer = ballerina // Error
+    dancer.perform()
+}
+```
+
+Calling `Dancer.perform` in generic code and on an `any Dancer` type
+from a main actor task
+is safe because it matches the isolation of the conformance.
+Calling `Dancer.perform` in generic code and on an `any Dancer` type
+from a concurrent task results in an error,
+because it would allow calling the main actor isolated implementation
+of `Dancer.perform` from outside the main actor.
+
+Abstract code that uses type parameters and `any` types
+can check whether a value conforms to a protocol
+through dynamic casting.
```

Comment:
This new part of the discussion needs a clearer connection to the current topic.  How is this different from the existing discussion of `as?` in the Type Casting chapter?  I think you're re-introducing that syntax as buildup for the next example, showing when that `as?` cast succeeds?